src/components/DataSession.vue

```diff
@@ -26,6 +27,39 @@ function addOperation(operationDefinition) {
     .catch(error => console.error('Error:', error));
 }
 
+const getImages = async () => {
+  try {
+    const response = await fetch ('http://127.0.0.1:8000/api/datasessions/', {
+      method: 'GET',
+      headers: {
+        'Authorization': 'Token 123456789abcdefg',
```

Comment:
As we start to add more requests to the backend API, we may want to think about using ajax with a pre-filter so that we can set the auth headers for every request in a single place, just to avoid repeated code and having to re-define the headers for each request. Here's an example: https://github.com/observatorycontrolsystem/science-archive-client/blob/d557d535092a004d9e17ba79df313ce9941d2ac7/src/main.js#L42C7-L42C7

Not sure if this is supported in fetch - googling around, it doesn't seem like it is but we could always create a util function that wraps the fetch call and adds in the auth info automatically

Reply:
we can also install axios : )

Reply:
This article explains what I'm thinking about nicely - I think it'd be good to add an axios interceptor for requests to the backend!

https://blog.bitsrc.io/setting-up-axios-interceptors-for-all-http-calls-in-an-application-71bc2c636e4e

Reply:
okey dokes! I'll read it 😎

Reply:
I'll make a story for this!

Reply:
Sounds good!!

Reply:
Sorry for responding late - the way I wrote the datasession is it gets all its data passed in through the data prop on creation - that should already contain the images and everything else it needs. You shouldn't need to do another get here, although we could change that for performance reasons in the future.